libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.h

```diff
@@ -23,11 +23,16 @@ class AP_BattMonitor_SMBus_Generic : public AP_BattMonitor_SMBus
 
     void timer(void) override;
 
-    // check if PEC supported with the version value in SpecificationInfo() function
-    // returns true once PEC is confirmed as working or not working
-    bool check_pec_support();
+    // Override capacity scaler, use the current multiplier from SpecificationInfo()
+    uint16_t get_capacity_scaler() const override { return _i_multiplier; }
 
-    uint8_t _pec_confirmed; // count of the number of times PEC has been confirmed as working
+    // Read SpecificationInfo() function, check if PEC is supported and update voltage and current multipliers
+    // returns true once SpecificationInfo() is returned from the Smart Battery
+    bool read_specification_info();
+
+    bool _specification_info_confirmed = false; // specification info successfully read
+    uint16_t _v_multiplier;    // voltage multiplier = 10^VScale
```

Comment:
I wonder if we should consider initialising these to "1" so that the multiplier will return 1 even before the read_specification_info() method has been called.